parastell/utils.py

```diff
@@ -202,3 +207,119 @@ def smooth_matrix(matrix, steps, sigma):
         previous_matrix = smoothed_matrix
 
     return smoothed_matrix
+
+
+class DAGMCMerger:
+    """Class to facilitate renumbering of entities to merge DAGMC models.
+
+    Arguments:
+        mb (PyMOAB Core): PyMOAB core with a DAGMC model loaded. Optional.
+    """
+
+    def __init__(self, mb=None):
+        self.mb = mb
+        if mb is None:
+            self.mb = core.Core()
+        self.geom_dim_tag = self.mb.tag_get_handle(
+            "GEOM_DIMENSION", 1, types.MB_TYPE_INTEGER, types.MB_TAG_DENSE
+        )
+        self.global_id_tag = self.mb.tag_get_handle(
+            "GLOBAL_ID", 1, types.MB_TYPE_INTEGER, types.MB_TAG_DENSE
+        )
+
+    def load_file(self, filename):
+        """Load DAGMC model from file.
+
+        Arguments:
+            filename (str): Path to DAGMC model to be loaded.
+        """
+        self.mb.load_file(filename)
+
+    def get_max_ids(self):
+        """Get the maximum id in the DAGMC model for each entity type.
+
+        returns:
+            max_ids (dict): {entity dimension (int): max_id (int)}
+        """
+        # dim 0: vertex
+        # dim 1: edge
+        # dim 2: surface
+        # dim 3: volume
+        # dim 4: group
+        max_ids = {dim: 0 for dim in range(5)}
+        root_set = self.mb.get_root_set()
+        geom_sets = self.mb.get_entities_by_type_and_tag(
+            root_set, types.MBENTITYSET, self.geom_dim_tag, [None]
+        )
```

Comment:
Why not do this in a loop of `dim` and get all the sets of each dim at once and grab the max:
```
for dim in range(5):
    dim_sets = self.mb.get_entities_by_type_and_tag(root_set, types.MBENTITYSET, self.geom_dim_tag, [dim])
    set_ids = self.mb.tag_get_data(self.global_id_tag, dim_sets)
    max_ids[dim] = max(set_ids)
```